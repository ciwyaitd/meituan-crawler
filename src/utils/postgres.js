'use strict'

const { Client } = require('pg')
const config = require('config')
const logger = require('../utils/logger')

if (!config.services.postgres.user) {
  throw new Error('Please set environment variable POSTGRES_USER')
}

if (!config.services.postgres.password) {
  throw new Error('Please set environment variable POSTGRES_PASSWORD')
}

if (!config.services.postgres.host) {
  throw new Error('Please set environment variable POSTGRES_HOST')
}

if (!config.services.postgres.port) {
  throw new Error('Please set environment variable POSTGRES_PORT')
}

if (!config.services.postgres.database) {
  throw new Error('Please set environment variable POSTGRES_DATABASE')
}

const client = new Client({
  user: config.services.postgres.user,
  password: config.services.postgres.password,
  host: config.services.postgres.host,
  port: config.services.postgres.port,
  database: config.services.postgres.database
})

client.connect()

client.on('error', (err) => {
  logger.error(`[postgres] Postgres has some error: ${err.stack}`)
})

client.on('end', () => {
  logger.info('[postgres] Postgres is end.')
  client.connect()
})

async function testConnect () {
  await client.query('SELECT 1')
  logger.info('[postgres] Postgres is ready.')
}

module.exports = {
  client,
  testConnect
}
