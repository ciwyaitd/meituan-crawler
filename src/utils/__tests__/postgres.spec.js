'use strict'

describe('utils.postgres', () => {
  beforeEach(() => jest.resetModules())
  it('should throw error when process.env.POSTGRES_USER is undefined', () => {
    expect.assertions(2)
    try {
      require('../postgres')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e).toMatchSnapshot()
    }
  })

  it('should throw error when process.env.POSTGRES_PASSWORD is undefined', () => {
    process.env.POSTGRES_USER = 'postgres'
    expect.assertions(2)
    try {
      require('../postgres')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e).toMatchSnapshot()
    }
  })

  it('should throw error when process.env.POSTGRES_HOST is undefined', () => {
    process.env.POSTGRES_USER = 'postgres'
    process.env.POSTGRES_PASSWORD = '123456'
    expect.assertions(2)
    try {
      require('../postgres')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e).toMatchSnapshot()
    }
  })

  it('should throw error when process.env.POSTGRES_PORT is undefined', () => {
    process.env.POSTGRES_USER = 'postgres'
    process.env.POSTGRES_PASSWORD = '123456'
    process.env.POSTGRES_HOST = 'localhost'
    expect.assertions(2)
    try {
      require('../postgres')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e).toMatchSnapshot()
    }
  })

  it('should throw error when process.env.POSTGRES_DATABASE is undefined', () => {
    process.env.POSTGRES_USER = 'postgres'
    process.env.POSTGRES_PASSWORD = '123456'
    process.env.POSTGRES_HOST = 'localhost'
    process.env.POSTGRES_PORT = 5432
    expect.assertions(2)
    try {
      require('../postgres')
    } catch (e) {
      expect(e).toBeInstanceOf(Error)
      expect(e).toMatchSnapshot()
    }
  })
})
