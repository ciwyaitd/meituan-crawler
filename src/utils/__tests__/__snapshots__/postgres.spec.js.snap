// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.postgres should throw error when process.env.POSTGRES_DATABASE is undefined 1`] = `[Error: Please set environment variable POSTGRES_DATABASE]`;

exports[`utils.postgres should throw error when process.env.POSTGRES_HOST is undefined 1`] = `[Error: Please set environment variable POSTGRES_HOST]`;

exports[`utils.postgres should throw error when process.env.POSTGRES_PASSWORD is undefined 1`] = `[Error: Please set environment variable POSTGRES_PASSWORD]`;

exports[`utils.postgres should throw error when process.env.POSTGRES_PORT is undefined 1`] = `[Error: Please set environment variable POSTGRES_PORT]`;

exports[`utils.postgres should throw error when process.env.POSTGRES_USER is undefined 1`] = `[Error: Please set environment variable POSTGRES_USER]`;
